rosetta-lang/src/main/java/com/regnosys/rosetta/cache/RequestScopedCacheModule.java

```diff
@@ -0,0 +1,64 @@
+package com.regnosys.rosetta.cache;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.inject.AbstractModule;
+import com.google.inject.TypeLiteral;
+import com.google.inject.multibindings.Multibinder;
+import com.regnosys.rosetta.cache.caches.ExpectedTypeCache;
+import com.regnosys.rosetta.cache.caches.ExpressionTypeCache;
+import com.regnosys.rosetta.cache.caches.RDataTypeCache;
+import com.regnosys.rosetta.cache.caches.RuleComputationCache;
+import com.regnosys.rosetta.cache.caches.RulesInputTypeCache;
+
+
+public class RequestScopedCacheModule extends AbstractModule {
+	private final List<IRequestScopedCache<?, ?>> caches = new ArrayList<>();
+	
+	private void configureCaches() {
+		caches.add(
+			new RDataTypeCache(
+				CacheBuilder.newBuilder()
+					.build()
+			));
+		
+		caches.add(
+			new ExpectedTypeCache(
+				CacheBuilder.newBuilder()
+					.build()
+			));
+		
+		caches.add(
+			new RuleComputationCache(
+				CacheBuilder.newBuilder()
+					.build()
+			));
+		
+		caches.add(
+			new RulesInputTypeCache(
+				CacheBuilder.newBuilder()
+					.build()
+			));
+
+		caches.add(
+			new ExpressionTypeCache(
+				CacheBuilder.newBuilder()
+					.weakKeys() // keys are reference-based and should be cleaned up under memory pressure
+					.build()
+			));
```

Comment:
The only caching configuration I have added for now is weak keys for the expression type inference. Not sure if there is anything else I should add.